src/etc/htmldocck.py

```diff
@@ -448,6 +478,28 @@ def check_command(c, cache):
                 ret = expected == found
             else:
                 raise InvalidCheck('Invalid number of @{} arguments'.format(c.cmd))
+
+        elif c.cmd == 'snapshot':  # snapshot test
+            if len(c.args) == 3:  # @snapshot <snapshot-name> <html-path> <xpath>
+                [snapshot_name, html_path, pattern] = c.args
```

Comment:
Is this pattern matching syntax supported on enough Python versions? I guess we'll find out once bors tests it...

Reply:
You should be able to test with python 2.7. we don't need to support anything besides 2.7 and "recent" versions of python 3 (3.4+ is probably fine)

Reply:
Actually, I don't think this is actually pattern matching, it's just destructuring, which should be supported even in ancient versions of python.

Reply:
Yep, just tested and this form of destructuring works on 2.7.